File: BeatsWave/Server/src/Data/BeatsWave.Data.Models/Beat.cs
namespace BeatsWave.Data.Models
{
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;

    using BeatsWave.Data.Common.Models;

    using static Validation.Beat;

    public class Beat : BaseDeletableModel<int>
    {
        public Beat()
        {
            this.Likes = new HashSet<Like>();
            this.Comments = new HashSet<BeatComment>();
        }

        [Required]
        [MaxLength(MaxNameLength)]
        public string Name { get; set; }

        [Required]
        public string BeatUrl { get; set; }

        [Required]
        public string ImageUrl { get; set; }

        public int Price { get; set; }

        [Required]
        public Genre Genre { get; set; }

        public int? Bpm { get; set; }

        [MaxLength(MaxDescriptionLength)]
        public string Description { get; set; }

        [Required]
        public string ProducerId { get; set; }

        [Required]
        public bool IsSold { get; set; }

        [Required]
        public int Plays { get; set; }

        public virtual ApplicationUser Producer { get; set; }

        public virtual ICollection<Like> Likes { get; }

        public virtual ICollection<BeatComment> Comments { get; set; }
    }
}

File: BeatsWave/Client/src/app/core/models/Beat.ts
export interface Beat {
    id: number,
    name: string,
    imageUrl: string,
    url: string,
    producerUserName: string,
    price: number,
    bpm : number,
    producerId: string,
    genre: string,
    description: string,
    likesCount: number,
    commentsCount: number,
    isLiked: boolean,
    createdOn: Date,
    producerProfileMainPhotoUrl: string,
    producerEmail: string,
    isProducerReceivingEmails: boolean
}
